parsel/csstranslator.py

```diff
@@ -89,6 +101,116 @@ def xpath_text_simple_pseudo_element(self, xpath):
         """Support selecting text nodes using ::text pseudo-element"""
         return XPathExpr.from_xpath(xpath, textnode=True)
 
+    def xpath_first_of_pseudo(self, xpath):
```

Comment:
cssselect allows properly extending only pseudo-elements (functional or not).
But these extensions here are more [structural pseudo-classes](https://www.w3.org/TR/css3-selectors/#pseudo-classes) than [pseudo-elements](https://www.w3.org/TR/css3-selectors/#pseudo-elements), hence the single `:`

> Pseudo-elements create abstractions about the document tree beyond those specified by the document language. For instance, document languages do not offer mechanisms to access the first letter or first line of an element's content. Pseudo-elements allow authors to refer to this otherwise inaccessible information. Pseudo-elements may also provide authors a way to refer to content that does not exist in the source document (e.g., the ::before and ::after pseudo-elements give access to generated content).
> A pseudo-element is made of two colons (::) followed by the name of the pseudo-element.
> This :: notation is introduced by the current document in order to establish a discrimination between pseudo-classes and pseudo-elements. For compatibility with existing style sheets, user agents must also accept the previous one-colon notation for pseudo-elements introduced in CSS levels 1 and 2 (namely, :first-line, :first-letter, :before and :after). This compatibility is not allowed for the new pseudo-elements introduced in this specification.

And consequently when using single colon `:`, cssselect is not interpreting it  as pseudo-elements or functional pseudo-elements, and the methods need to be defined for the translation to work in cssselect.

So this PR is indeed more of a hack than a proper extension of CSS selectors with cssselect.

We may want to work on making cssselect extensible also for pseudo-classes.
Would you want to go that route @eliasdorneles ?